dao/story.dao.server.js

```diff
@@ -67,6 +67,8 @@ const unflagStory = (story, userID) => {
     }
     return null;
 };
+const getSortedFlagged = (numberOfStories) =>
+    storyModel.find({ $where: "this.flagged_by_users.length > 0" }).sort({flagged_by_users: 'desc'}).limit(parseInt(numberOfStories));
```

Comment:
I am not able to get descending order based on number of flagged when testing out locally on Postman.
It seems here the sort adds the user_ids of the arrays and does descending according to that. For example I tested stories with following two arrays. And array1 was shown higher than array2 inspite of array2 having more flags. I think this is because of the number 100000000 in array1.
array1 = 
"flagged_by_users": [
            666,
            6663,
            6664,
            45,
            100000000
        ]

array2 =
 "flagged_by_users": [
            6664,
            89,
            8,
            23,
            45,
            10,
            11,
            12
        ]
I then put 200000000 in array2 and when I did that array2 was shown as bigger. It seems here directly calling sort({flagged_by_users: 'desc'} is adding the elements in the array and then sorting based on that, it is not sorting based on the length of the array

Reply:
nice eye, will look into it